cookbooks/aws-parallelcluster-platform/test/controls/disable_sudo_access_spec.rb

```diff
@@ -0,0 +1,50 @@
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+control 'tag:config_disable_sudo_access_disabed_action' do
+  title 'Check if sudo access for default user is disabled'
+  only_if { node['cluster']["disable_sudo_access_for_default_user"] == 'true' }
+
+  describe file('/etc/sudoers') do
+    it { should exist }
+    its('content') { should_not match %r{\b\S*(#{node['cluster']['cluster_user']} ALL=(ALL) NOPASSWD:ALL)} }
+  end
+
+  describe file('/etc/sudoers.d/99-parallelcluster-revoke-sudo-access') do
+    it { should exist }
+    its('owner') { should eq 'root' }
+    its('group') { should eq 'root' }
+    its('mode') { should cmp '0600' }
+    its('content') { should match %r{#{node['cluster']['cluster_user']} ALL=(ALL) !ALL} }
+  end
+
+end
+
+
+control 'tag:config_disable_sudo_access_enabled_action' do
```

Comment:
Controls that starts with `tag:config` are automatically executed as kitchen tests in Jenkins (see [readme](https://github.com/aws/aws-parallelcluster-cookbook?tab=readme-ov-file#kitchen-tests-definition)).
It means that you cannot have both "disabled_action" and "enabled_action" with `tag:config`.

You can use `tag:config` only for tests you expect to have them passing if executed at instance bootstrap (kitchen tests in Jenkins). I'd say only "disabled" can have `tag:config` prefix because this is the default behaviour.

Reply:
Agreed but my `only_if { node['cluster']["disable_sudo_access_for_default_user"] == 'false' }` should be able to Skip the other test